corehq/apps/events/views.py

```diff
@@ -236,8 +245,86 @@ def dispatch(self, *args, **kwargs):
             raise Http404
         return super(AttendeesListView, self).dispatch(*args, **kwargs)
 
-    def post(self, *args, **kwargs):
-        return super(AttendeesListView, self).post(*args, **kwargs)
+    @property
+    @memoized
+    def new_attendee_form(self):
+        if self.request.method == "POST":
+            return NewAttendeeForm(self.request.POST)
+        return NewAttendeeForm()
+
+    @property
+    def page_context(self):
+        return {
+            'new_attendee_form': self.new_attendee_form,
+        }
+
+    @allow_remote_invocation
```

Comment:
@kaapstorm Just curious why you decided to go with the remote method invocation strategy and not the (arguably more standard) remote procedure call?

(I'm not really familiar with RMI apart from knowing what it is, but I'm still naively wondering if a POST wouldn't have been simpler?)

Reply:
I basically reproduced the way we create mobile workers.

I think there is _a lot_ here that can be simplified. (Getting `stagedAttendee` vs `newAttendee` working was a pain in the ass, because I kept on adding code that I had hoped I could leave out.)

I am very interested in finding out out how best to simplify this. If we decide to add an `external_id` field, and check for uniqueness, then I'll refer to how we do that for `username` for mobile workers, and then see how much we can pull out and refactor. -- At least, those were my thoughts, but I'm very keen to hear other ways to keep this as simple as possible.